src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -48,10 +54,44 @@ object WorkflowActor {
         case Failure(ex) => Future.failed(ex)
       }
     }
+
+    override def start(actor: WorkflowActor, actorRef: ActorRef) = actor.startRunnableCalls()
   }
   
   case object Restart extends WorkflowActorMessage with StartMode {
-    override def runInitialization(actor: WorkflowActor): Future[Unit] = Future.successful(())
+
+    override def runInitialization(actor: WorkflowActor): Future[Unit] = {
+      for {
+        _ <- actor.backend.prepareForRestart(actor.workflow)
+        _ <- actor.dumpTables()
+      } yield ()
+    }
+
+    override def start(actor: WorkflowActor, actorRef: ActorRef) = {
```

Comment:
Why do we need both of these parameters?  Is it not possible to get the `ActorRef` from the `WorkflowActor`?